services/geocoding-v6.js

```diff
@@ -0,0 +1,231 @@
+ *  Options are [IETF language tags](https://en.wikipedia.org/wiki/IETF_language_tag) comprised of a mandatory
+ *  [ISO 639-1 language code](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) and optionally one or more IETF subtags for country or script.
+ * @param {String} [config.address_number] - The number associated with the house (Stuctured Input specific field)
+ * @param {String} [config.street] - The name of the street in the address (Stuctured Input specific field)
+ * @param {String} [config.place] - Typically these are cities, villages, municipalities, etc. (Stuctured Input specific field)
+ * @param {String} [config.region] - Top-level sub-national administrative features, such as states in the United States or provinces in Canada or China. (Stuctured Input specific field)
+ * @param {String} [config.country] - The number associated with the house (Stuctured Input specific field)
+ * @param {String} [config.postcode] - Postal codes used in country-specific national addressing systems. (Stuctured Input specific field)
+ * @param {String} [config.locality] - Official sub-city features (Stuctured Input specific field)
+ * @param {boolean} [config.autocomplete=true] - Return autocomplete results or not.
+ * @param {String} [config.worldview="us"] - Filter results to geographic features whose characteristics are defined differently by audiences belonging to various regional, cultural, or political groups.
+ * @return {MapiRequest}
+ *
+ * @example
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   limit: 2
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ * 
+ * @example
+ * // geocoding in structured input mode
+ * geocodingClient.forwardGeocode({
+ *   mode: 'structured',
+ *   address_number: '12',
+ *   street: 'Main str.'
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ *
+ * @example
+ * // geocoding with proximity
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   proximity: [-95.4431142, 33.6875431]
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ *
+ * // geocoding with countries
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   countries: ['fr']
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ *
+ * // geocoding with bounding box
+ * geocodingClient.forwardGeocode({
+ *   query: 'Paris, France',
+ *   bbox: [2.14, 48.72, 2.55, 48.96]
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const match = response.body;
+ *   });
+ */
+GeocodingV6.forwardGeocode = function(config) {
+  v.assertShape({
+    query: v.required(v.string),
+    mode: v.oneOf('normal', 'structured'),
+    countries: v.arrayOf(v.string),
+    proximity: v.oneOf(v.coordinates, 'ip'),
```

Comment:
Can you check what object-type is referred to by `v.coordinates`? i.e. does it expect a number array, or does it accept an object with `latitude` and `longitude` properties?

Reply:
it's an array of numbers `v.tuple(v.number, v.number)`